sympy/sets/sets.py

```diff
@@ -994,6 +993,29 @@ def _eval_imageset(self, f):
             return
         var = f.variables[0]
 
+        if expr.is_Piecewise:
+            result = S.EmptySet
+            set_value = self
```

Comment:
This name feels confusing as 'value' is usually associated with the output of a function while here we are in its domain of definition.